plugins/org.eclipse.n4js.generator.headless/src/org/eclipse/n4js/generator/headless/N4HeadlessCompiler.java

```diff
@@ -165,21 +162,25 @@ public int compare(IN4JSProject o1, IN4JSProject o2) {
 	 * Private constructor to prevent accidental instantiation.
 	 */
 	@Inject
-	private N4HeadlessCompiler(CompositeGenerator compositeGenerator, JavaIoFileSystemAccess fsa) {
-		this.compositeGenerator = compositeGenerator;
+	private N4HeadlessCompiler(JavaIoFileSystemAccess fsa) {
 		this.fsa = fsa;
-		this.outputs = buildInitialOutputConfiguration();
-		fsa.setOutputConfigurations(this.outputs);
 	}
 
-	private Map<String, OutputConfiguration> buildInitialOutputConfiguration() {
+	/** Build an output configuration from a composite generator. */
+	private Map<String, OutputConfiguration> buildOutputConfigurations(ICompositeGenerator compGenerator) {
 		Map<String, OutputConfiguration> result = new HashMap<>();
-		for (CompilerDescriptor desc : compositeGenerator.getCompilerDescriptors()) {
+		for (CompilerDescriptor desc : compGenerator.getCompilerDescriptors()) {
 			result.put(desc.getIdentifier(), desc.getOutputConfiguration());
 		}
 		return result;
 	}
 
+	/** Configure FileSystemAccess (FSA) from a composite generator and a project */
+	private void configureFSAOutput(ICompositeGenerator compGenerator, IN4JSProject project) {
```

Comment:
also: move this method next to method `configureFSA`

Reply:
Done. Removed this method.